src/hotspot/cpu/riscv/templateInterpreterGenerator_riscv.cpp

```diff
@@ -1098,7 +1098,22 @@ address TemplateInterpreterGenerator::generate_native_entry(bool synchronized) {
     __ ld(x28, Address(xmethod, Method::native_function_offset()));
     address unsatisfied = (SharedRuntime::native_method_throw_unsatisfied_link_error_entry());
     __ mv(t1, unsatisfied);
-    __ ld(t1, Address(t1, 0));
+    if (AvoidUnalignedAccesses)
+    {
+      __ mv(t, t1);
+      __ lhu(t1, Address(t1, 0));
+      __ lhu(t0, Address(t, 2));
+      __ slli(t0, t0, 16);
+      __ add(t1, t1, t0);
+      __ lhu(t0, Address(t, 4));
+      __ slli(t0, t0, 32);
+      __ add(t1, t1, t0);
+      __ lhu(t0, Address(t, 6));
+      __ slli(t0, t0, 48);
+      __ add(t1, t1, t0);
+    } else {
+      __ ld(t1, Address(t1, 0)); //2 bytes alligned, but not 4 or 8
```

Comment:
```suggestion
      __ ld(t1, Address(t1, 0)); // 2 bytes alligned, but not 4 or 8
```
Should we put this comment into `if` statement?